usb: cdc_acm: log related warning promoted to error

**Describe the bug**
It seems that a `cdc_acm` logging feature is triggering a warning that gets promoted to error in twister.

**To Reproduce**
Steps to reproduce the behavior:
1. `twister -i -p degu_evk -s zephyr/tests/posix/common/portability.posix.common.tls.newlib`
2. See error

**Expected behavior**
The test should build and pass.

**Impact**
What impact does this issue have on your progress (e.g., annoyance, showstopper)

**Logs and console output**
```shell
/__w/zephyr/zephyr/subsys/usb/device/class/cdc_acm.c
/__w/zephyr/zephyr/subsys/usb/device/class/cdc_acm.c:63:2: error: #warning "USB_CDC_ACM_LOG_LEVEL forced to LOG_LEVEL_NONE" [-Werror=cpp]
   63 | #warning "USB_CDC_ACM_LOG_LEVEL forced to LOG_LEVEL_NONE"
      |  ^~~~~~~
cc1: all warnings being treated as errors
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows): any
 - Toolchain (e.g Zephyr SDK, ...): Zephyr SDK 0.15.2
 - Commit SHA or Version used: 1173db6f605aca79f38d6fbcf2e634f5dbd02f3f

**Additional context**
It's likely that this has to do with a recent change in logging cc @nordic-krch 

This is currently failing in #52977 even after a rebase.

--- a/boards/arm/degu_evk/Kconfig.defconfig
+++ b/boards/arm/degu_evk/Kconfig.defconfig
@@ -21,4 +21,16 @@ config UART_LINE_CTRL
 
 endif # USB_DEVICE_STACK
 
+if LOG
+
+# Logger cannot use itself to log
+config USB_CDC_ACM_LOG_LEVEL
+	default 0
+
+# Set USB log level to error only
+config USB_DEVICE_LOG_LEVEL
+	default 1
+
+endif # LOG
+
 endif # BOARD_DEGU_EVK

